lib/board.rb

```diff
@@ -29,4 +29,50 @@ def valid_coordinate?(coordinate)
       true
     end
   end
+
+  def valid_placement?(ship, coordinates)
+      if ship.length == coordinates.length && coordinates.all? { |coordinate| valid_coordinate? (coordinate) }
+        if place_ship?(coordinates) == true && is_occupied?(coordinates) == false
+          true
+        else
+          false
+        end
+      else
+        false
+      end
+    end
+
+    def is_occupied?(coordinates)
+    coordinates.any? { |coordinate| !@cells[coordinate].empty? }
+  end
+
+  def place_horizontal?(coordinates)
+    coordinates.each_cons(2).all? do |coordinate1, coordinate2|
+      coordinate1_letter = coordinate1[0].ord #A of A1
+      coordinate1_number = coordinate1[1].to_i #1 of A1
+
+      coordinate2_letter = coordinate2[0].ord
+      coordinate2_number = coordinate2[1].to_i
+
+      coordinate1_letter == coordinate2_letter &&
+      coordinate1_number == (coordinate2_number - 1)
+    end
+  end
+
+  def place_vertical?(coordinates)
+    coordinates.each_cons(2).all? do |coordinate1, coordinate2|
+      coordinate1_letter = coordinate1[0].ord #A of A1
```

Comment:
I left these comments in for reference because this was tricky for me to wrap my head around.